### PR TITLE
SA-3001: Get/Set-JCCloudDirectory

### PR DESCRIPTION
**Pull request by gweinjc**
## Issues
* [SA-3001](https://jumpcloud.atlassian.net/browse/SA-3001) - Get/Set-JCCloudDirectory

## What does this solve?
Adds two new functions to the PowerShell module, `Get-JCCloudDirectory` and `Set-JCCloudDirectory`

The `Get-JCCloudDirectory` function will allow admins to query their Cloud Directory instances for their configurations as well as seeing what users or usergroups are associated

The `Set-JCCloudDirectory` function will allow admins to set the properties for the Cloud Directory instances. In it's current iteration, the Set-JCCloudDirectory function does not allow the association of users or usergroups

## Is there anything particularly tricky?
N/A

## How should this be tested?
Pester test suites have been written. Manual tests would look like the following:

### Get-JCCloudDirectory

- `Get-JCCloudDirectory -Type <g_suite/office_365>` -- Returns all configured directories of specified type
- `Get-JCCloudDirectory -Name <DirectoryName> `-- Returns directory by Name
- `Get-JCCloudDirectory -ID <DirectoryID> `-- Returns directory by ID
- `Get-JCCloudDirectory -Name <DirectoryName> -Association Users `– Returns user associations for the specified directory
- `Get-JCCloudDirectory -Name <DirectoryName> -Association UserGroups `– Returns UserGroup associations for the specified directory

### Set-JCCloudDirectory

- `Set-JCCloudDirectory -Name <DirectoryName> -GroupsEnabled $false `– Disables security groups for that directory
- `Set-JCCloudDirectory -Name <DirectoryName> -UserLockoutAction 'suspend' `– Sets the UserLockoutAction for that directory
- `Set-JCCloudDirectory -Name <DirectoryName> -UserPasswordExpirationAction 'suspend' `– Sets the UserPasswordExpirationAction for that directory
- `Set-JCCloudDirectory -Name <DirectoryName> -NewName ”NewName” `– Renames the directory in console

[SA-3001]: https://jumpcloud.atlassian.net/browse/SA-3001?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: PowerShell/JumpCloud Module/Public/Directory/Set-JCCloudDirectory.ps1 [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            $body.GetEnumerator() | ForEach-Object {
+                $message = '{0} {1}' -f $_.key, $_.value
+                Write-Debug $message
+            }
+        }
+
+        if ($CloudDirectory.Type -eq 'office_365') {
+            $resultsArray = Update-JcSdkOffice365 -Office365Id $CloudDirectory.Id -Body $body
+        } elseif ($CloudDirectory.Type -eq 'g_suite') {
+            $resultsArray = Update-JcSdkGSuite -Id $CloudDirectory.Id -Body $body
```
Comment (jworkmanjc) [inline PowerShell/JumpCloud Module/Public/Directory/Set-JCCloudDirectory.ps1:76]:
```suggestion
            $resultsArray = Update-JcSdkGSuite -Office365Id $CloudDirectory.Id -Body $body
```

Comment (jworkmanjc) [reply PowerShell/JumpCloud Module/Public/Directory/Set-JCCloudDirectory.ps1:76]: SDKs were updated looks like the parameter changed?

**Review by jworkmanjc — CHANGES_REQUESTED** [review]
Nice work here, last bit of the puzzle I'm trying to square away is why I can't revert changes to the `groupsEnabled` parameter when editing an `office365` object. I want us to figure that out but otherwise this all looks good. 

Just one change with the ID parameter to make and let's see if we can solve the issue with `groupsEnabled` before calling this good.

File: PowerShell/JumpCloud Module/Public/Directory/Set-JCCloudDirectory.ps1 [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                if (!$CloudDirectory) {
+                    throw "$ID was not found. Please try again"
+                }
+            }
+        }
+
+        if ($NewName) {
+            $body.Add('name', $NewName)
+        }
+        if ($GroupsEnabled) {
```
Comment (jworkmanjc) [inline PowerShell/JumpCloud Module/Public/Directory/Set-JCCloudDirectory.ps1:52]:
```suggestion
        if (-not [System.String]::IsNullOrEmpty($GroupsEnabled)) {
```

Comment (jworkmanjc) [reply PowerShell/JumpCloud Module/Public/Directory/Set-JCCloudDirectory.ps1:52]: If we type `Set-JCCloudDirectory -Name "Office365" -GroupsEnabled $false` well, $groupsEnabled is always going to be false and we'll never enter this block of code.

File: PowerShell/JumpCloud Module/Public/Directory/Set-JCCloudDirectory.ps1 (hunk 1 above)
Comment (gweinjc) [reply PowerShell/JumpCloud Module/Public/Directory/Set-JCCloudDirectory.ps1:76]: The Gsuite SDK is just -Id whereas the Office365 SDK is -Office365Id

**Review by kmaranionjc — DISMISSED** [review]
Tested good on my side. Nice work!

**Review by jworkmanjc — APPROVED** [review]
Thanks for the detailed work on examples, this looks good.

**Review by kmaranionjc — APPROVED** [review]

**Pull request merged**